### PR TITLE
Add Course Navigation Component

### LINKED ISSUE #98: Create Base Course, Unit, Lesson Page Component
**Is your feature request related to a problem? Please describe.**
We need to create a base drawer component, which will be used for creating units, lessons, and words.

Screen Shot 2022-03-10 at 8 40 17 PM

Subject to change, so check [Figma](https://www.figma.com/file/4jXpfEsFIe0KCgZKHllxBm/7000-Languages-Design?node-id=22%3A77) for the latest design 

**Describe the solution you'd like**

Based on current Figma designs, there are 3 pages that look the same: Contributor/Course Home, Contributor/Unit, and Contributor/Lesson. All 4 drawer components have some similarities, like a title at the top, and some differences, like the actual data being presented.

We can create a base component that all 3 pages will implement. The component can have the following:
1. A title
2. A back button
3. Description text at the top of the screen
4. Cards in the body of the text
5. ....

One way to breakdown this issue is to split it up into small components. For example:
1. Start with setting the title and description text
2. Create a simple card component
3. Add the rest of the components, like the Add New button.

Proper use of React props will be critical here. Some of the props required are:
- string props for setting the text on the screen
- .....
The rest are for you to determine.

**Tips**

While creating the entire tab is daunting, I suggest breaking down the issue into smaller problems.

Remember to take small steps, and if you think that this issue will take some time, communicate it and put up multiple PRs for this issue 🤩

### PR DESCRIPTION
**Pull request by exrlla**
### Status
:construction: In development

### Description
This PR covers the creation of the course home component, which will be used to make the rest of the `Contributor/Unit` page and the `Contributor/Lesson` page. 

Fixes #98 

## Todos
- finish component 
- style accordingly `Units` page
- style accordingly `Lesson` page

## Screenshots
Screen Shot 2022-03-10 at 8 40 17 PM

### CODE REVIEW

File: client/src/components/CourseNavHome/CourseNavHome.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
+import React from 'react'
```
Comment (exrlla) [inline client/src/components/CourseNavHome/CourseNavHome.js:1]: rename file to `Language Home`  index.js, Folder, and `ManageView` to `languagehome` adn the import in `CourseHome.js`

File: client/src/components/CourseNavHome/CourseNavHome.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    <Text
+    style={{
+        fontFamily: 'GT_Haptik_bold',
+        
+      }}
+      fontSize={23}
+      paddingTop={3}
+      paddingLeft={5}
+      >    
+{unitsNumber} 4 Units
```
Comment (exrlla) [inline client/src/components/CourseNavHome/CourseNavHome.js:157]: number of "units" or "lessons"... is determined from length of data, name of `Units `is a prop

File: client/src/components/CourseNavHome/CourseNavHome.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    languageDescription,
+    unitsNumber,
+    buttonText,
+    rightIconName,
+    unitsName,
+    lessonNumber,
+    statusTag,
+
+  })
+  const data = [
```
Comment (exrlla) [inline client/src/components/CourseNavHome/CourseNavHome.js:47]: add data to the 'CourseHome' page to clean up code!

File: client/src/components/CourseNavHome/CourseNavHome.js [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    <StyledButton
+        title={buttonText}
+        variant="manage"
+        fontSize={15}
+        rightIcon={<MaterialCommunityIcons
+        name={rightIconName}
+        color={colors.red.dark}
+        size={20}
+        > </MaterialCommunityIcons>}
+        onPress={() => navigation.navigate('ManageUnits')}
```
Comment (exrlla) [inline client/src/components/CourseNavHome/CourseNavHome.js:168]: create new prop to pass in a string for the onPress function

File: client/src/components/CourseNavHome/CourseNavHome.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        display: 'flex',
+        borderBottomLeftRadius: 20,
+        borderBottomRightRadius: 20
+    },
+    manageBar: {
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'space-between'
+    },
+    selected: {
```
Comment (exrlla) [inline client/src/components/CourseNavHome/CourseNavHome.js:32]: delete

**Review by exrlla — COMMENTED** [review]
glhf

**Review by ashayp22 — APPROVED** [review]
stupendous

**Pull request merged**